examples/spmd/loader.c

```diff
@@ -33,13 +33,38 @@
 #include <bsg_manycore_regression.h>
 
 #include <stdio.h>
-#include <sys/ioctl.h>
+//#include <sys/ioctl.h>
 #include <unistd.h>
 #include <string.h>
 #include <time.h>
 #include <stdlib.h>
 #include <sys/stat.h>
 
+/*
+ * Reads manycore specific symbols from the program binary
+ * @param[out] program: Pointer to a memory location containing the address of the manycore binary
+ * @param[out] program_size: Pointer to the manycore binary size
+ * @returns HB_MC_SUCCESS
+ */ 
+static int read_symbols(unsigned char **program, size_t *program_size) {
+        unsigned char *address;
+        uint64_t size;
+
+        // Read the address from the symbol table in the binary
+        __asm__ __volatile__ ("la %[manycore_start_addr], manycore"
+                             : [manycore_start_addr] "=r" (address));
+
+        // The program size is already stored at a predefined location in the binary
+        __asm__ __volatile__ ("la t0, manycore_size\n\t"
+                              "ld %[manycore_size], 0(t0)"
+                             : [manycore_size] "=r" (size));
+
+        *program_size = size;
+        *program = address;
+
+        return HB_MC_SUCCESS;
+}
+
```

Comment:
This function contains the C code to read the symbols from the program binary. Let me know how you guys want to go ahead (i.e. leave it and make it BP-specific somehow or throw it or keep this PR open until the FS stuff is figured out)